src/lib.rs

```diff
@@ -0,0 +1,198 @@
+//! application logic of the blockchain (the _Runtime_) is encoded as a Wasm module, and is stored
+//! on-chain. Other than the runtime, the rest of the system is called the "client" software, which
+//! is a native binary capable of doing all the redundant, non-application-specific work, such as
+//! networking, consensus, and database management.
+//!
+//! To learn more about the substrate architecture using some visuals, see [`substrate_diagram`].
+//!
+//! All in all, this design enables all substrate-based chains to achieve forkless, self-enacting
+//! upgrades out of the box. Combined with governance abilities that are shipped with `FRAME`, this
+//! enables a chain to always evolve.
+//!
+//! ## How to Get Stared
+//!
+//! Most developers want to leave the client side code as-is, and focus on the runtime. To do so,
+//! look into the [`frame_support`] crate, which is the entry point crate into runtime development.
+//!
+//! > Side note, it is entirely possible to craft a substrate-based runtime without FRAME, an
+//! > example of which can be found [here](https://github.com/JoshOrndorff/frameless-node-template).
+//!
+//! In more broad terms, two common avenues exist into substrate:
+//!
+//! 1. Starting with templates: A number of substrate-based templates exist and they can be used for
+//!    various purposes, with zero to little additional code needed. All of these templates contain
+//!    runtimes that are highly configurable and are likely suitable for basic needs.
+//! 2. Customizing the client: To the contrary, some developers may want to customize the client
+//!    side software to achieve novel goals such as a new consensus engine, or a new database
+//!    backend. While Substrate's main configurability is in the runtime, the client is also highly
+//!    generic and can be customized to a great extent.
+//!
+//! ## Structure
+//!
+//! Substrate is a massive cargo workspace with hundreds of crates, therefore it is useful to know
+//! how to navigate its crates.
+//!
+//! In broad terms, it is divided into three categories:
+//!
+//! * `sc-*` (short for *substrate-client*) crates, located under `./client` folder. These are all
+//!   the client crates. Notable examples are crates such as [`sc-network`], various consensus
+//!   crates, [`sc-rpc-api`] and [`sc-client-db`], all of which are expected to reside in the client
+//!   side.
+//! * `sp` (short for *substrate-primitives*) crates, located under `./primitives` folder. These are
+//!   the traits that glue the client and runtime together, but are not opinionated about what
+//!   framework is using for building the runtime. Notable examples are [`sp-api`] and [`sp-io`],
+//!   which form the communication bridge between the client and runtime, as explained in
+//!   [`substrate_diagram`].
+//! * `pallet-*` and `frame-*` crates, located under `./frame` folder. These are the crates related
+//!   to FRAME. See [`frame_support`] for more information.
```

Comment:
Some idea:  

Maybe mention the `utils` folder containing all kinds of helper binaries.  
So Substrate is in this way self contained and has little additional dependencies, besides `scale`, `type-info` and `bounded-collections`.